File: backend/app/schemas/kitchen_rotation.py
"""
주방 로테이션 미리보기/저장 API 데이터 형태.

  GET /api/kitchen-rotation?year=&month=  전월 기준 자동 계산(이미 저장돼 있으면 그 값) 반환
  PUT /api/kitchen-rotation               사장님이 확인/수정한 값을 그 달 기준으로 저장
"""

from pydantic import BaseModel, Field


class KitchenRotationItem(BaseModel):
    staff_id: int
    staff_name: str
    position: str = Field(pattern="^(open|mid|close)$")


class KitchenRotationPreview(BaseModel):
    year: int
    month: int
    items: list[KitchenRotationItem]
    saved: bool  # 이미 이 달 값으로 저장된 적 있으면 True, 아니면 방금 계산만 한 미리보기


class KitchenRotationPutItem(BaseModel):
    staff_id: int
    position: str = Field(pattern="^(open|mid|close)$")


class KitchenRotationPut(BaseModel):
    year: int
    month: int
    items: list[KitchenRotationPutItem]

File: frontend/src/api/kitchenRotation.ts
// 주방 오픈/미들/마감 3인 월별 로테이션 미리보기/저장 ("근무표 관리" 자동배치 화면).
import { apiGet, apiSend } from "./client";

export type RotationSlot = "open" | "mid" | "close";

export interface RotationItem {
  staff_id: number;
  staff_name: string;
  position: RotationSlot;
}

export interface RotationPreview {
  year: number;
  month: number;
  items: RotationItem[];
  saved: boolean; // false = 전월 기준으로 방금 계산만 한 미리보기
}

export const SLOT_LABEL: Record<RotationSlot, string> = {
  open: "오픈",
  mid: "미들",
  close: "마감",
};

export const getRotation = (year: number, month: number) =>
  apiGet<RotationPreview>(`/kitchen-rotation?year=${year}&month=${month}`);

export const saveRotation = (
  year: number,
  month: number,
  items: { staff_id: number; position: RotationSlot }[],
) => apiSend<RotationPreview>("PUT", "/kitchen-rotation", { year, month, items });
